test/bin_test.py

```diff
@@ -32,4 +38,11 @@ class LuigidDaemonTest(server_test.ServerTestRun):
 
     @mock.patch('daemon.DaemonContext')
     def run_server(self, daemon_context):
-        luigi.cmdline.luigid(['--port', str(self._api_port), '--background', '--logdir', '.', '--pidfile', self.id() + '.pid'])
+        tempdir = tempfile.mkdtemp(prefix=self.id())
```

Comment:
It's difficult to use a context manager with unittest's method call interface, especially in this case where this is an overridden method